Adapter/MoPubAdapterMediatedNativeAd.m

```diff
@@ -1,151 +0,0 @@
-
-@end
-
-@implementation MoPubAdapterMediatedNativeAd
-
-- (instancetype)initWithMoPubNativeAd:
-        (nonnull MPNativeAd *)moPubNativeAd mappedImages: (NSMutableDictionary *)downloadedImages {
-    
-  if (!moPubNativeAd) {
-    return nil;
-  }
-
-  self = [super init];
-  if (self) {
-    _nativeAd = moPubNativeAd;
-    _nativeAdProperties = moPubNativeAd.properties;
-
-    //rupa fill the extras
-    CGFloat defaultImageScale = 1;
-    
-      if(downloadedImages!=nil){
-          _mappedImages = [[NSArray alloc] initWithObjects:[downloadedImages objectForKey:kAdMainImageKey], nil];
-          _mappedLogo = [downloadedImages objectForKey:kAdIconImageKey];
-          
-      }
-      else{
-          NSURL *mainImageUrl = [[NSURL alloc] initFileURLWithPath:[self.nativeAdProperties objectForKey:kAdMainImageKey]];
-          _mappedImages = @[ [[GADNativeAdImage alloc] initWithURL:mainImageUrl scale:defaultImageScale] ];
-
-          NSURL *logoImageURL = [[NSURL alloc] initFileURLWithPath:[self.nativeAdProperties objectForKey:kAdIconImageKey]];
-          _mappedLogo = [[GADNativeAdImage alloc] initWithURL:logoImageURL scale:defaultImageScale];
-      }
-    
-  }
-  return self;
-}
-
-- (NSString *)headline {
-  return [self.nativeAdProperties objectForKey:kAdTitleKey];
-}
-
-- (NSString *)body {
-  return [self.nativeAdProperties objectForKey:kAdTextKey];
-}
-
-- (NSArray *)images {
-  return self.mappedImages;
-}
-
-- (GADNativeAdImage *)logo {
-  return self.mappedLogo;
-}
-
-- (NSString *)callToAction {
-  return [self.nativeAdProperties objectForKey:kAdCTATextKey];
-}
-
-- (NSString *)advertiser {
-    return nil;
-}
-
-- (NSDictionary *)extraAssets {
-  return self.extras;
-}
-
-- (id<GADMediatedNativeAdDelegate>)mediatedNativeAdDelegate {
-  return self;
-}
-
-#pragma mark - GADMediatedNativeAdDelegate implementation
-
-// Because the Sample SDK handles click and impression tracking via methods on its native
-// ad object, there's no need to pass it a reference to the UIView being used to display
-// the native ad. So there's no need to implement mediatedNativeAd:didRenderInView here.
-// If your mediated network does need a reference to the view, this method can be used to
-// provide one.
-
-- (void)mediatedNativeAd:(id<GADMediatedNativeAd>)mediatedNativeAd
-         didRenderInView:(UIView *)view viewController:(UIViewController *)viewController;
-{
-    [_nativeAd performSelector:@selector(willAttachToView:) withObject:view];
-}
-
-- (void)mediatedNativeAdDidRecordImpression:(id<GADMediatedNativeAd>)mediatedNativeAd {
-  if (self.nativeAd) {
-  }
-}
-
-- (void)mediatedNativeAd:(id<GADMediatedNativeAd>)mediatedNativeAd
-    didRecordClickOnAssetWithName:(NSString *)assetName
-                             view:(UIView *)view
-                   viewController:(UIViewController *)viewController {
-  if (self.nativeAd) {
-      [_nativeAd performSelector:@selector(trackClick)];
```

Comment:
This line is not necessary [_nativeAd performSelector:@selector(trackClick)]; as adViewTapped will call trackClick.